### PR TITLE
Ensure consistency when rendering the sent event indicator

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/17937

The main change to how we find the last successful event we sent is in the first commit, the remaining work is tests & ensuring all code paths pass the newly calculated value around

Before
image

After
image

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Ensure consistency when rendering the sent event indicator ([\#11314](https://github.com/matrix-org/matrix-react-sdk/pull/11314)). Fixes vector-im/element-web#17937.

### CODE REVIEW

File: src/components/structures/MessagePanel.tsx [hunk 1]
```diff
@@ -608,6 +613,11 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         }
     }
 
+    private isSentState(ev: MatrixEvent): boolean {
+        const status = ev.getAssociatedStatus();
+        return !status || status === EventStatus.SENT;
```
Comment (weeman1337) [inline src/components/structures/MessagePanel.tsx:618]: Why is a falsy status considered as a sent state (function name)?
Can you add a comment?

File: src/components/structures/MessagePanel.tsx [hunk 2]
```diff
@@ -1069,25 +1069,26 @@ export default class MessagePanel extends React.Component<IProps, IState> {
  * Holds on to an event, caching the information about whether it should be
```
Comment (weeman1337) [inline src/components/structures/MessagePanel.tsx:1069]: Can you update the docs here?

**Review by weeman1337 — APPROVED** [review]
Looks good to me. :+1:  It would be great if you could look at the comments about the doc.

File: src/components/structures/MessagePanel.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/MessagePanel.tsx:618]: image

That's just how MatrixEvent works, the status is only ever of outgoing events, and once we get the remote-echo we clear the status

**Comment by MTRNord** [comment]
This seem to have caused a small regression where it doesnt disappear anymore when the person read it but sent themself something while reading:

image

The upper checkmark is on my message and the readmarker is on the event sent by the other person

**Pull request merged**